ssl/statem/statem.c

```diff
@@ -157,13 +157,8 @@ int ossl_statem_skip_early_data(SSL *s)
     if (s->ext.early_data != SSL_EARLY_DATA_REJECTED)
         return 0;
 
-    if (s->hello_retry_request) {
-        if (s->statem.hand_state != TLS_ST_SW_HELLO_RETRY_REQUEST)
```

Comment:
Well, in fact, the server sends the HRR *immediately* after processing the ClientHello. So from a server's perspective *all* early data records arrive afterwards (as if they had still be in flight). This PR is all about skipping early data records after HRR. We now handle it in a manner consistent with how we handle early data if there is no HRR, i.e. we enter the special `TLS_ST_EARLY_DATA` state. This means that the ossl_statem_skip_early_data() function no longer needs to special case an HRR.